department-of-veterans-affairs/vets-website
# Prevent modal content overflow

## Are you removing, renaming or moving a folder in this PR?

## Did you change site-wide styles, platform utilities or other infrastructure?

## Summary
Add a max length to middle name on the full name schema. Use modal text overrides. Ensure long words break and do not overflow on the Pension Benefits form (21P-527EZ).

## Issue
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/93814

## Related issue(s)
- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/3355

## Associated Pull Request(s)
- https://github.com/department-of-veterans-affairs/vets-json-schema/pull/944

## How to run in local environment
1. Check out this branch locally
2. Run `vets-website` and `vets-api`
3. In your local environment, set the `pension_multiple_page_response` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_multiple_page_response
4. Go to `http://localhost:3001/pension/apply-for-veteran-pension-form-21p-527ez/` in your browser

## How to verify
1. Go to the '/military/other-names/summary' page
2. Select 'Yes' to add a previous name
3. Enter a first, middle, and last name that is the max 30 characters in length
4. Continue to the 'summary' page
5. Verify the 'Remove' modal content does not overflow
6. Verify the 'Cancel' modal content does not overflow when cancelling an entry
7. Verify the 'Alert's content width does not overflow

## What areas of the site does it impact?
Pension Benefits Application

## Screenshots
### Desktop
| Before      | Delete       | Cancel    | Alert      |
| ----------- | ----------- | ----------- | ----------- |
|![image](https://github.com/user-attachments/assets/67762ab7-10ab-4c8d-a09f-1a4ba92c745c)|![Screenshot 2024-10-11 at 8 30 35 AM](https://github.com/user-attachments/assets/aae51468-d0c1-46c7-82d9-ea0a188925c0)|![Screenshot 2024-10-11 at 8 31 08 AM](https://github.com/user-attachments/assets/4b5a4362-b931-4e84-81bd-538866583be0)|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_introduction (7)](https://github.com/user-attachments/assets/7347ae10-705b-4557-8700-cada4610cca6)

### Mobile (if applicable)
| Before      | After       |
| ----------- | ----------- |
|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_resume(iPhone XR)](https://github.com/user-attachments/assets/65c52ef5-576f-44a2-bb11-62821daf2485)|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_form-saved(iPhone XR)](https://github.com/user-attachments/assets/6aa413df-101c-4d67-94fc-640a9c46bb0d)


### Quality Assurance & Testing

### Error Handling

### Authentication

## Requested Feedback
(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_
